components/molecules/cards/card-event/card-event.types.ts

```diff
@@ -0,0 +1,26 @@
+import { ComponentPropsWithoutRef, ElementType, ReactNode } from "react";
+
+import { ButtonPort } from "components/atoms/button/button.types";
+import { TagPort } from "components/atoms/tag";
+import { TIcon } from "components/layout/icon/icon.types";
+
+interface Variants {}
+
+interface ClassNames {
+  base: string;
+}
+
+export type CardEventDisplay = "planned" | "terminated" | "highlight";
+
+export interface CardEventPort<C extends ElementType> extends Partial<Variants> {
+  as?: C;
+  htmlProps?: ComponentPropsWithoutRef<C>;
+  classNames?: Partial<ClassNames>;
+  title: ReactNode;
+  titleIcon?: TIcon.Props;
+  text?: ReactNode;
+  tag?: TagPort<"span">;
+  display?: CardEventDisplay;
+  primaryAction?: ButtonPort<"button">;
+  secondaryAction?: ButtonPort<"button">;
```

Comment:
```suggestion
  titleProps: ReactNode;
  titleIconProps?: TIcon.Props;
  text?: ReactNode;
  tagProps?: TagPort<"span">;
  status?: CardEventDisplay;
  primaryActionProps?: ButtonPort<"button">;
  secondaryActionProps?: ButtonPort<"button">;
```
Maybe prefer naming the props like this ? I haven't updated the types here, it's just to show the idea. We can discuss this.